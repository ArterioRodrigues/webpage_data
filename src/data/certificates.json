{
  "sectionTitle": "certifications",
  "sectionSubtitle": "Certificates that I have collected over time",
  "content": [
    {
      "name": "Responsive Web Design Certification",
      "url": "https://www.freecodecamp.org/certification/caglarturali/responsive-web-design"
    },
    {
      "name": "Introduction to Game Development",
      "url": "https://www.coursera.org/account/accomplishments/certificate/GKDBNAGKPD6M"
    },
    {
      "name": "Introduction to Virtual Reality",
      "url": "https://www.coursera.org/account/accomplishments/certificate/UC3L666SD5TG"
    },
    {
      "name": "3D Models for Virtual Reality",
      "url": "https://www.coursera.org/account/accomplishments/certificate/4YHK3LYWMJPV"
    },
    {
      "name": "The Art of Music Production",
      "url": "https://www.coursera.org/account/accomplishments/certificate/DNVYLQQL3WVF"
    },
    {
      "name": "The Technology of Music Production",
      "url": "https://www.coursera.org/account/accomplishments/certificate/VZDUNEC24XER"
    },
    {
      "name": "Pro Tools Basics",
      "url": "https://www.coursera.org/account/accomplishments/certificate/YW4QY26UHC5F"
    }
  ]
}
