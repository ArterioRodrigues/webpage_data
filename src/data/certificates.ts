import { Page, Certificate, Linkable } from '../models';

export const certificates: Page<Certificate<Linkable>> = {
  name: 'certifications',
  desc: 'Certificates that I have earned over time',
  content: [
    {
      name: 'Full-Stack Web Development with React Specialization',
      url:
        'https://www.coursera.org/account/accomplishments/specialization/certificate/XGPU5C68VAEE',
      totalCourses: 4,
      earned: [
        {
          name: 'Front-End Web UI Frameworks and Tools: Bootstrap 4',
          url:
            'https://www.coursera.org/account/accomplishments/certificate/TQ2HAVM6JJ73',
        },
        {
          name: 'Front-End Web Development with React',
          url:
            'https://www.coursera.org/account/accomplishments/certificate/QJPWRV28SCV3',
        },
        {
          name: 'Multiplatform Mobile App Development with React Native',
          url:
            'https://www.coursera.org/account/accomplishments/certificate/GNX5WEXJWSGY',
        },
        {
          name: 'Server-Side Development with NodeJS, Express and MongoDB',
          url:
            'https://www.coursera.org/account/accomplishments/certificate/XGMAYMHEXDZW',
        },
      ],
    },
    {
      name: 'freeCodeCamp | The Full Stack Development Certificate',
      url: 'https://guide.freecodecamp.org/certifications/',
      totalCourses: 6,
      earned: [
        {
          name: 'Responsive Web Design Certification',
          url:
            'https://www.freecodecamp.org/certification/caglarturali/responsive-web-design',
        },
        {
          name: 'JavaScript Algorithms and Data Structures Certification',
          url:
            'https://www.freecodecamp.org/certification/caglarturali/javascript-algorithms-and-data-structures',
        },
      ],
    },
    {
      name: 'Game Design and Development Specialization',
      url: 'https://www.coursera.org/specializations/game-development',
      totalCourses: 5,
      earned: [
        {
          name: 'Introduction to Game Development',
          url:
            'https://www.coursera.org/account/accomplishments/certificate/GKDBNAGKPD6M',
        },
      ],
    },
    {
      name: 'Virtual Reality Specialization',
      url: 'https://www.coursera.org/specializations/virtual-reality',
      totalCourses: 5,
      earned: [
        {
          name: 'Introduction to Virtual Reality',
          url:
            'https://www.coursera.org/account/accomplishments/certificate/UC3L666SD5TG',
        },
        {
          name: '3D Models for Virtual Reality',
          url:
            'https://www.coursera.org/account/accomplishments/certificate/4YHK3LYWMJPV',
        },
      ],
    },
    {
      name: 'Music Production Specialization',
      url: 'https://www.coursera.org/specializations/music-production',
      totalCourses: 4,
      earned: [
        {
          name: 'The Art of Music Production',
          url:
            'https://www.coursera.org/account/accomplishments/certificate/DNVYLQQL3WVF',
        },
        {
          name: 'The Technology of Music Production',
          url:
            'https://www.coursera.org/account/accomplishments/certificate/VZDUNEC24XER',
        },
        {
          name: 'Pro Tools Basics',
          url:
            'https://www.coursera.org/account/accomplishments/certificate/YW4QY26UHC5F',
        },
      ],
    },
  ],
};
