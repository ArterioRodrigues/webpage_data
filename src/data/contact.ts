import { ContactItem, Page } from '../models';

export const contact: Page<ContactItem> = {
  name: 'contact',
  desc: 'Feel free to contact me',
  content: [
    {
      name: 'Telegram',
      url: 'https://t.me/caglarturali',
      icon: ['fab', 'telegram-plane'],
      isMain: true,
    },
    {
      name: 'GitHub',
      url: 'https://github.com/caglarturali',
      icon: ['fab', 'github'],
    },
    {
      name: 'LinkedIn',
      url: 'https://linkedin.com/in/caglarturali',
      icon: ['fab', 'linkedin'],
    },
    {
      name: 'Instagram',
      url: 'https://www.instagram.com/caglarturali/',
      icon: ['fab', 'instagram'],
    },
    {
      name: 'goodreads',
      url: 'https://www.goodreads.com/caglarturali',
      icon: ['fab', 'goodreads'],
    },
  ],
};
