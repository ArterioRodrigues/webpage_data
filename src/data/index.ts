export * from './certificates';
export * from './contact';
export * from './education';
export * from './greeter';
export * from './metadata';
export * from './projects';
export * from './skills';
