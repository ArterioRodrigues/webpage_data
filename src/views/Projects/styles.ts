/**
 * Projects component styles.
 */
import { Styles } from 'react-jss';
import { mSize, orange, cyan, textDark } from '../../theme';

export default {
  entry: {
    marginBottom: '30px',
    '&>div': {
      marginBottom: '4px',
    },
  },
  name: {
    fontSize: mSize,
    fontWeight: 500,
    marginBottom: '2px',
  },
  description: {
    color: orange,
    fontWeight: 500,
    marginBottom: '2px',
  },
  tech: {
    color: cyan,
    fontWeight: 500,
  },
  homepage: {
    color: textDark,
  },
} as Styles;
