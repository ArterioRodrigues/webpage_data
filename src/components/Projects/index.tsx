/**
 * Projects component.
 */
import React from 'react';
import { createUseStyles } from 'react-jss';
import Container from '../Container';
import { Page, Project } from '../../models';
import styles from './styles';

const useStyles = createUseStyles(styles);

export interface ProjectsProps {
  data: Page<Project>;
}

const Projects: React.FC<ProjectsProps> = ({ data }) => {
  const classes = useStyles();

  return (
    <Container title={data.name} subtitle={data.desc}>
      <ul>
        {data.content.map(({ name, summary, tech, url }) => (
          <div className="container__list" key={name}>
            <li key={name} className={classes.entry}>
              <a href={url} target="_blank" rel="noopener noreferrer">
                <div className={classes.name}>{name}</div>
                <div className={classes.description}>{summary}</div>
                <div className={classes.tech}>{tech.join(', ')}</div>
              </a>
            </li>
          </div>
        ))}
      </ul>
    </Container>
  );
};

export default Projects;
