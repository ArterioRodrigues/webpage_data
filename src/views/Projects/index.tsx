/**
 * Projects component.
 */
import React from 'react';
import { createUseStyles } from 'react-jss';
import Container from '../../components/Container';
import { Project, BaseProps } from '../../models';
import styles from './styles';

const useStyles = createUseStyles(styles);

export type ProjectsProps = BaseProps<Project>;

const Projects: React.FC<ProjectsProps> = ({ data, fileName }) => {
  const classes = useStyles();

  return (
    <Container
      title={data.name}
      subtitle={data.desc}
      seo={{ title: fileName, description: data.desc }}
    >
      <ul>
        {data.content.map(({ name, summary, tech, url }) => (
          <div className="container__list" key={name}>
            <li key={name} className={classes.entry}>
              <a href={url} target="_blank" rel="noopener noreferrer">
                <div className={classes.name}>{name}</div>
                <div className={classes.description}>{summary}</div>
                <div className={classes.tech}>{tech.join(', ')}</div>
              </a>
            </li>
          </div>
        ))}
      </ul>
    </Container>
  );
};

export default Projects;
