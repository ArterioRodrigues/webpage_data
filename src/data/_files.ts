import { NavLink } from '../models';
import Greeter from '../views/Greeter';
import Skills from '../views/Skills';
import Education from '../views/Education';
import Projects from '../views/Projects';
import Certificates from '../views/Certificates';
import Contact from '../views/Contact';
import { certificates, contact, education, projects, skills } from '.';

export const files: NavLink[] = [
  {
    icon: ['fab', 'html5'],
    color: '#e44d26',
    url: '/',
    name: 'index.html',
    route: {
      component: Greeter,
      data: contact,
    },
  },
  {
    icon: ['fab', 'js'],
    color: '#ffca28',
    url: '/skills',
    name: 'skills.js',
    route: {
      component: Skills,
      data: skills,
    },
  },
  {
    icon: 'university',
    color: '#42a5f5',
    url: '/education',
    name: '.educationrc',
    route: {
      component: Education,
      data: education,
    },
  },
  {
    icon: 'sliders-h',
    color: '#98c379',
    url: '/projects',
    name: 'projects.config',
    route: {
      component: Projects,
      data: projects,
    },
  },
  {
    icon: ['fab', 'sass'],
    color: '#e13e76',
    url: '/certifications',
    name: 'certifications.sass',
    route: {
      component: Certificates,
      data: certificates,
    },
  },
  {
    icon: ['fab', 'markdown'],
    color: '#abb2bf',
    url: '/contact',
    name: 'contact.md',
    route: {
      component: Contact,
      data: contact,
    },
  },
];
