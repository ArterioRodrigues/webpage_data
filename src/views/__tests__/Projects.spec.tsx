import React from 'react';
import { render } from '@testing-library/react';
import Projects from '../Projects';
import { projects } from '../../data';

describe('<Projects />', () => {
  test('should render correctly', () => {
    const { asFragment } = render(
      <Projects data={projects} fileName="projects.test" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
