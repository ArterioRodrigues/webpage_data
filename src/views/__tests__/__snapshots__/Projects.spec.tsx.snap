// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Projects /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="root-0-2-6"
  >
    <h2
      class="title-0-2-7"
    >
      projects
    </h2>
    <h3
      class="subtitle-0-2-8"
    >
      Some of the projects I've worked on so far
    </h3>
    <div
      class="content-0-2-9"
    >
      <ul>
        <div
          class="container__list"
        >
          <li
            class="entry-0-2-1"
          >
            <a
              href="https://getkernel.now.sh/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div
                class="name-0-2-2"
              >
                GetKernel
              </div>
              <div
                class="description-0-2-3"
              >
                The kernel store for Ubuntu derivatives.
              </div>
              <div
                class="tech-0-2-4"
              >
                NextJS
              </div>
            </a>
          </li>
        </div>
        <div
          class="container__list"
        >
          <li
            class="entry-0-2-1"
          >
            <a
              href="https://turali.js.org/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div
                class="name-0-2-2"
              >
                turali.js.org
              </div>
              <div
                class="description-0-2-3"
              >
                The web app you are currently looking at.
              </div>
              <div
                class="tech-0-2-4"
              >
                React, TypeScript
              </div>
            </a>
          </li>
        </div>
        <div
          class="container__list"
        >
          <li
            class="entry-0-2-1"
          >
            <a
              href="https://turali.js.org/girizgah/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div
                class="name-0-2-2"
              >
                Girizgah
              </div>
              <div
                class="description-0-2-3"
              >
                LightDM Webkit theme for Pardus Linux.
              </div>
              <div
                class="tech-0-2-4"
              >
                React
              </div>
            </a>
          </li>
        </div>
        <div
          class="container__list"
        >
          <li
            class="entry-0-2-1"
          >
            <a
              href="https://deepintr.org/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div
                class="name-0-2-2"
              >
                deepintr.org
              </div>
              <div
                class="description-0-2-3"
              >
                Landing page for an online community.
              </div>
              <div
                class="tech-0-2-4"
              >
                GatsbyJS, React, TypeScript
              </div>
            </a>
          </li>
        </div>
        <div
          class="container__list"
        >
          <li
            class="entry-0-2-1"
          >
            <a
              href="https://github.com/caglarturali/WatcherCatBot/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div
                class="name-0-2-2"
              >
                WatcherCatBot
              </div>
              <div
                class="description-0-2-3"
              >
                A Telegram bot that returns popularity info of the given Linux distribution.
              </div>
              <div
                class="tech-0-2-4"
              >
                Node.js, Express.js
              </div>
            </a>
          </li>
        </div>
        <div
          class="container__list"
        >
          <li
            class="entry-0-2-1"
          >
            <a
              href="https://play.google.com/store/apps/details?id=com.astroloci.app"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div
                class="name-0-2-2"
              >
                Astroloci
              </div>
              <div
                class="description-0-2-3"
              >
                A horoscope application for Android.
              </div>
              <div
                class="tech-0-2-4"
              >
                Native Android App, Express.js REST API
              </div>
            </a>
          </li>
        </div>
        <div
          class="container__list"
        >
          <li
            class="entry-0-2-1"
          >
            <a
              href="https://play.google.com/store/apps/details?id=com.caglarturali.bilenadam"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div
                class="name-0-2-2"
              >
                Bilen Adam
              </div>
              <div
                class="description-0-2-3"
              >
                Lottery machine for Turkey's national lottery games.
              </div>
              <div
                class="tech-0-2-4"
              >
                Unity 3D
              </div>
            </a>
          </li>
        </div>
        <div
          class="container__list"
        >
          <li
            class="entry-0-2-1"
          >
            <a
              href="https://play.google.com/store/apps/details?id=com.caglarturali.the2o48"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div
                class="name-0-2-2"
              >
                2‚óè48
              </div>
              <div
                class="description-0-2-3"
              >
                A clone of the popular puzzle game 2048, with unlimited levels.
              </div>
              <div
                class="tech-0-2-4"
              >
                Unity 3D
              </div>
            </a>
          </li>
        </div>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
